iot-hub/Quickstarts/device-streams-proxy/device/DeviceStreamSample.cs

```diff
@@ -51,12 +51,23 @@ private static async Task HandleOutgoingDataAsync(NetworkStream localStream, Cli
 
         public async Task RunSampleAsync()
         {
-            await RunSampleAsync(true).ConfigureAwait(false);
+            while (true)
```

Comment:
I think the best way is to:
- expose this cancellation token as an argument of RunSampleAsync, and
- have the infinite loop but check if the cancellation token has been triggered;
- on Main invoke RunSampleAsync passing a cancellation token, but without a timer.

that way the sample becomes safer in case a customer gets it as is to use in production, having a way to escape the infinite loop.

But it would be good to get some input from @CIPop

Reply:
@ewertons, done. Have a look in the new committed version